wire-runtime/src/main/java/com/squareup/wire/MessageAdapter.java

```diff
@@ -501,14 +501,14 @@ private void writeValueNoTag(WireOutput output, Object value, Datatype datatype)
       case BOOL: output.writeRawByte((Boolean) value ? 1 : 0); break;
       case ENUM: writeEnum((ProtoEnum) value, output); break;
       case STRING:
-        final byte[] bytes = ((String) value).getBytes("UTF-8");
-        output.writeVarint32(bytes.length);
+        ByteString bytes = ByteString.encodeUtf8((String) value);
```

Comment:
Actually it does. It uses the platform's built-in encoder without unnecessary extra allocations.